webapp/src/shared/hooks/UseBlock.tsx

```diff
@@ -1,24 +1,69 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useCallback, useEffect, useReducer } from 'react';
 import { ResponseError, UserResponseDto } from '../generated';
 import { usersApi } from '../services/ApiService';
 
+export type BlockRelation = {
+  isUserBlocked: boolean;
+  amIBlocked: boolean;
+};
+
+type BlockAction =
+  | { type: 'FILL_BLOCK_RELATION'; isUserBlocked: boolean; amIBlocked: boolean }
+  | { type: 'BLOCK_USER' }
+  | { type: 'UNBLOCK_USER' };
+
+const blockReducerCallback = (
+  state: BlockRelation | null,
+  action: BlockAction,
+): BlockRelation | null => {
+  const { type } = action;
+
+  if (type === 'FILL_BLOCK_RELATION') {
+    return {
+      isUserBlocked: action.isUserBlocked,
+      amIBlocked: action.amIBlocked,
+    };
+  }
+
+  if (!state) {
+    return null;
+  }
+
+  if (type === 'BLOCK_USER') {
+    return {
+      ...state,
+      isUserBlocked: true,
+    };
+  } else if (type === 'UNBLOCK_USER') {
+    return {
+      ...state,
+      isUserBlocked: false,
+    };
+  }
+  throw new Error('Unknown block reducer action');
+};
+
 export function useBlock(user: UserResponseDto | null) {
-  const [isBlocked, setIsBlocked] = useState<boolean | null>(null);
+  const [blockRelation, dispatch] = useReducer(blockReducerCallback, null);
```

Comment:
Since we start to have a more complex way to handle this state, better to use a semantic dispatch